pkg/apis/certmanager/v1alpha1/types_certificate.go

```diff
@@ -101,6 +101,12 @@ type CertificateSpec struct {
 	// key size of 256 will be used for "ecdsa" key algorithm and
 	// key size of 2048 will be used for "rsa" key algorithm.
 	KeyAlgorithm KeyAlgorithm `json:"keyAlgorithm,omitempty"`
+
+	// KeyEncoding is the private key cryptography standards (PKCS) number
+	// for this certificate's private key to be encoded in. If provided, allowed
+	// values are 1 and 8 standing for PKCS#1 and PKCS#8, respectively.
+	// If KeyEncoding is not specified, then PKCS#1 will be used by default.
+	KeyEncoding int `json:"keyEncoding,omitempty"`
```

Comment:
I think we should consider what kinds of values we accept as part of our API carefully for this.

I'm not too familiar with the various different ways you can encode a private key, but what other formats might we want in future here? It seems like restricting input to a simple integer could limit us in future.

Using a string value with an associated set of consts (similar to KeyAlgorithm in the same value) would make extensibility in future easier.